engine/classes/Elgg/EntityIconService.php

```
@@ -435,6 +435,11 @@ public function getSizes($entity_type = null, $entity_subtype = null, $type = 'i
 			$sizes = $this->hooks->trigger("entity:$type:sizes", $entity_type, $params, $sizes);
 		}
 
+		if (empty($sizes)) {
```

Maybe just throw an exception if not an array